firedrake/functionspaceimpl.py

```diff
@@ -308,6 +348,54 @@ def boundary_nodes(self, sub_domain):
     def collapse(self):
         return type(self).create(self.topological.collapse(), self.mesh())
 
+    @classmethod
+    def make_function_space(cls, mesh, element, name=None):
+        r"""Factory method for :class:`.WithGeometryBase`."""
+        mesh.init()
+        topology = mesh.topology
+        if isinstance(element, finat.ufl.MixedElement) and not isinstance(element, (finat.ufl.VectorElement, finat.ufl.TensorElement)):
+            spaces = [cls.make_function_space(topology, e) for e in element.sub_elements]
+            new = MixedFunctionSpace(spaces, name=name)
+        else:
+            # Check that any Vector/Tensor/Mixed modifiers are outermost.
+            check_element(element)
+            if element.family() == "Real":
+                new = RealFunctionSpace(topology, element, name=name)
+            else:
+                new = FunctionSpace(topology, element, name=name)
+        if mesh is not topology:
+            new = cls.create(new, mesh)
+        return new
+
+    def reconstruct(self, mesh=None, name=None, **kwargs):
+        r"""Reconstruct this :class:`.WithGeometryBase` on a different mesh."""
```

Comment:
Please provide a more detailed docstring and make sure to state what `**kwargs` are used for.